lib/hx/hx_v2.ml

```diff
@@ -0,0 +1,447 @@
+open Base
+open Tyxml_html
+
+module Verb = struct
+  type t =
+    [ `Delete of Uri.t
+    | `Get of Uri.t
+    | `Patch of Uri.t
+    | `Post of Uri.t
+    | `Put of Uri.t
+    ]
+
+  let delete_attr uri = Uri.to_string uri |> Unsafe.string_attrib "hx-delete"
+  let get_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-get"
+  let patch_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-patch"
+  let post_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-post"
+  let put_attr uri = uri |> Uri.to_string |> Unsafe.string_attrib "hx-put"
```

Comment:
don't think we need these functions?

can just do the match in `to_attr` to get the attr type maybe

Reply:
These are hidden via the mli file, but I'm using them in the `Attributes` module to expose functions for each verb that can directly take a `Uri.t` rather than taking a variant (though I also have `verb` which will take the variant versions).

That way we can use the `Attributes` module like so in our components + TyXML:

```ocaml
let _ = 
  let endpoint = Uri.of_string "/foo_bar" in
  let button = Button.make ~attrs:[Hx.Attributes.get endpoint]
  ...
  ```
  
  rather than:
  ```ocaml
let _ = 
  let endpoint = Uri.of_string "/foo_bar" in
  let button = Button.make ~attrs:[Hx.Attributes.verb (Hx.Verb.Get endpoint)]
  ...
  ```

Reply:
Rereading this I think I misunderstood your comment - I've updated the code to removes those fns and just reuse `to_attrs` 😅